weblate/addons/forms.py

```diff
@@ -46,6 +46,27 @@ def save(self):
         return self._addon.instance
 
 
+class GenerateMoForm(BaseAddonForm):
+    path = forms.CharField(
+        label=_('Path of generated MO file'),
+        required=False,
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(GenerateMoForm, self).__init__(*args, **kwargs)
+        self.helper = FormHelper(self)
+        self.helper.layout = Layout(
+            Field('path'),
+        )
+
+    def test_render(self, value):
+        translation = self._addon.instance.component.translation_set.all()[0]
+        validate_render(value, translation=translation)
+
+    def clean_path(self):
+        self.test_render(self.cleaned_data['path'])
```

Comment:
Is this really needed? You don't seem to use rendering in the addon code, so there is no need to validate the markup here as it won't get used.

Reply:
On the other side it should use `validate_filename` to avoid use of `..` in the path.

Reply:
I'm not sure about those two parts if I'm honest.  I just took them from one of the other addons and tweaked.  I totally expected that there may be something i've overlooked or a test that needs to be implemented that isn't there because I'm still learning how you code the system 👍

Reply:
Oh and the .. is required if you are relative to the PO files.  I guess we should check that you are not trying to go outside of the component root's absolute path though?

Reply:
How would that work with linked_repo's as I have noticed those mentioned in the code a few times.

Reply:
You don't have to care about linked repos here, it's transparent on this level.